source/usage-examples/changeStream.txt

```diff
@@ -184,3 +166,62 @@ If you run the preceding example, you should see the following output:
 
       const changeStream = collection.watch(pipeline, options);
 
+Listener Function
+~~~~~~~~~~~~~~~~~
+
+The following example opens a change stream on the ``haikus`` collection in
+the ``insertDB`` database. Let's create a listener function to receive and
+print change events that occur on the collection.
+
+First, open the change stream on the collection and then define a listener
+on the change stream using the ``on()`` method. Once set, generate a change
+event by performing a change to the collection.
+
+To generate the change event on the collection, let's use ``insertOne()``
+method to add a new document. Since the ``insertOne()`` may run before the
+listener function can register, we use a timer, defined as
+``simulateAsyncPause`` to wait 1 second before executing the insert.
+
+We also use ``simulateAsyncPause`` after the insertion of the document
+to provide ample time for the listener function to receive the change
+event and for the listener to complete its execution before
+closing the ``ChangeStream`` instance using the ``close()`` method.
+
+The timers used in this example are only necessary for this demonstration
+to make sure there is enough time to register listener and have the
+listener process the event before exiting.
```

Comment:
S: Put this in a Note admonition to make it stand out

```suggestion
.. note:: You don't need timers to use listener functions.

   The timers in this code example are for demonstration purposes only.
```

...or something similar.

Reply:
I'm in favor of deleting the redundant note.